### PR TITLE
Fix/button rerender

### PR DESCRIPTION
**Pull request by jamesducky**
Closes[BUG PlmgButton - Rendering of dynamic text](https://app.asana.com/0/1200489836971088/1203428670027960/f)

### Summary of changes included in this PR

- Enables shadow dom.
- Adds form submit and reset events to replace component scoped events. 
- Tests for events
-  Sets the host component to the size of button content

### Reviewer checklist:

- Tests for
  - accessibility
  - rendered HTML (spec)
- Storybook stories for all variants
- JSDoc documentation for component
- Update Example app (React, ...)

### CODE REVIEW

File: packages/component-library/src/components/plmg-button/plmg-button.tsx [hunk 1]
```diff
@@ -250,40 +278,44 @@ export class Button {
...
+            target={this.target}
+            aria-label={this.label}
+          >
+            {this.hasIconLeft() && (
+              <plmg-svg-icon class={'icon-left'} icon={this.iconLeft} />
+            )}
+            {this.hasIconCenter() && (
+              <plmg-svg-icon class={'icon-center'} icon={this.iconCenter} />
+            )}
+            <slot />
```
Comment (TanguyDucky) [inline packages/component-library/src/components/plmg-button/plmg-button.tsx:295]:
```suggestion
            <slot></slot>
```
Safer to not use self-closing tags

File: packages/component-library/src/components/plmg-button/plmg-button.tsx [hunk 2]
```diff
@@ -250,40 +278,44 @@ export class Button {
...
+      <Host style={{ width: this.fullWidth ? 'full-width' : 'fit-content' }}>
+        <button class={classes} type={this.type} aria-label={this.label}>
           {this.hasIconLeft() && (
             <plmg-svg-icon class={'icon-left'} icon={this.iconLeft} />
           )}
           {this.hasIconCenter() && (
             <plmg-svg-icon class={'icon-center'} icon={this.iconCenter} />
           )}
-          <slot></slot>
+          <slot />
```
Comment (TanguyDucky) [inline packages/component-library/src/components/plmg-button/plmg-button.tsx:313]:
```suggestion
          <slot></slot>
```

File: packages/component-library/src/components/plmg-button/plmg-button.tsx (hunk 1 above)
Comment (jamesducky) [reply packages/component-library/src/components/plmg-button/plmg-button.tsx:295]: added

File: packages/component-library/src/components/plmg-button/plmg-button.tsx (hunk 2 above)
Comment (jamesducky) [reply packages/component-library/src/components/plmg-button/plmg-button.tsx:313]: added

**Review by TanguyDucky — APPROVED** [review]
Works great on the React example app.

It's a bit of a hacky way around shadow DOM limitations, of course. But I guess we're using bleeding edge tech here so we need to compromise a little bit

**Pull request merged**